api/src/main/java/org/openmrs/api/db/hibernate/HibernateUserDAO.java

```diff
@@ -122,12 +123,21 @@ public User getUserByEmail(String email) {
 	 */
 	@Override
 	@SuppressWarnings("unchecked")
-	public User getUserByActivationKey(String activationKey) {
-		User user = (User) sessionFactory.getCurrentSession().createCriteria(User.class).add(Restrictions.like("activationKey", activationKey, MatchMode.START)).uniqueResult();	
-		if(user != null) {
-			String[] tokens = user.getActivationKey().split(":");
-			if(tokens[0].equals(activationKey)) {
-				return user;
+	public User getUserByActivationKey(String token) {
+		LoginCredential loginCred = getLoginCredentialByToken(token);
+		if(loginCred != null) {
+			User user = getUser(loginCred.getUserId());
```

Comment:
please combine these into one line

Reply:
It always pointless to assign a value to a new variable when on the next line you return, should always just combine it into a a single statement that returns